packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3085,8 +3085,8 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
 
   /// The configuration for this element.
   @override
-  Widget get widget => _widget;
-  Widget _widget;
+  Widget get widget => _widget!;
+  Widget? _widget;
```

Comment:
I think this is necessary to reduce the severity of memory leaks.
It would be good to test what the impact on performance of this change is.
Fyi @leafpetersen it would be nice if the compiler could detect that these null checks always succeed for the typical case when the element is known to have not been disposed.

Reply:
I'm not sure how much the compiler could really optimize this. We're promising to not need this unless `_widget` is non-null, but to verify that there'd need to be a nullcheck on it. That's basically what's happening here, right?

Reply:
> It would be good to test what the impact on performance of this change is.

Yeah, I am also curious to see if this has any impact on our performance benchmarks.